atomic_reactor/util.py

```diff
@@ -852,4 +852,18 @@ def df_parser(df_path, workflow=None, cache_content=False, env_replace=True, par
     return dfparser
 
 
+def is_valid_tag_name(tag):
+    """Check if tag name of an image is valid
 
+    See <https://docs.docker.com/engine/reference/commandline/tag/>
+
+        A tag name must be valid ASCII and may contain lowercase and
+        uppercase letters, digits, underscores, periods and dashes.
+        A tag name may not start with a period or a dash and may contain
+        a maximum of 128 characters.
+    """
+    PATTERN = r'^[a-zA-Z0-9_][a-zA-Z0-9_.-]{0,127}$'
```

Comment:
Let's use `\w` here since it's the equivalent to `[a-zA-Z0-9_]`: `r'[\w][\w.-]{0,127}$`

https://docs.python.org/2/library/re.html